### PR TITLE
Collate consecutive heights and sequence numbers shown in logs

### PR DESCRIPTION
**Pull request by romac**
Closes: #2846

## Description

Collate consecutive sequence numbers and heights in logs

For example, where before we would only show the first 50 sequence numbers:

> 2022-11-09T09:49:59.911813Z DEBUG ThreadId(386) worker.batch{chain=ibc-1}:supervisor.handle_batch{chain=ibc-1}:supervisor.process_batch{chain=ibc-1}:worker.packet.cmd{src_chain=ibc-1 src_port=transfer src_channel=channel-5 dst_chain=ibc-0}:schedule_packet_clearing{height=Some(Height { revision: 1, height: 26800 })}:relay_pending_packets{height=Some(Height { revision: 1, height: 26799 })}:build_packet_ack_msgs{query_height=1-26799}: sequence numbers of ack packets to send to the destination chain out of the ones with acknowledgments on the source chain (first 50 shown here) dst_chain=ibc-0 src_chain=ibc-1 total=184 sequences=[1259, 1260, 1261, 1262, 1263, 1264, 1265, 1266, 1267, 1268, 1269, 1270, 1271, 1272, 1273, 1274, 1275, 1276, 1277, 1278, 1279, 1280, 1281, 1282, 1283, 1284, 1285, 1286, 1287, 1288, 1289, 1290, 1291, 1292, 1293, 1294, 1295, 1296, 1297, 1298, 1299, 1300, 1301, 1302, 1303, 1304, 1305, 1306, 1307, 1308]

We now show all sequences numbers where consecutive ones are collated together:

> 2022-11-09T09:49:59.911813Z DEBUG ThreadId(386) worker.batch{chain=ibc-1}:supervisor.handle_batch{chain=ibc-1}:supervisor.process_batch{chain=ibc-1}:worker.packet.cmd{src_chain=ibc-1 src_port=transfer src_channel=channel-5 dst_chain=ibc-0}:schedule_packet_clearing{height=Some(Height { revision: 1, height: 26800 })}:relay_pending_packets{height=Some(Height { revision: 1, height: 26799 })}:build_packet_ack_msgs{query_height=1-26799}: sequence numbers of ack packets to send to the destination chain out of the ones with acknowledgments on the source chain dst_chain=ibc-0 src_chain=ibc-1 total=184 sequences=1259..=1329, 1359..=1439, 1469..=1500

### PR author checklist:
- [x] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [x] Added tests: integration (for Hermes) or unit/mock tests (for modules).
- [x] Linked to GitHub issue.
- [x] Updated code comments and documentation (e.g., `docs/`).
- [x] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: crates/relayer-cli/src/commands/query/packet/acks.rs [hunk 1]
```diff
@@ -72,8 +65,11 @@ impl QueryPacketAcknowledgementsCmd {
 // cargo run --bin hermes -- query packet acknowledgements --chain ibc-0 --port transfer --connection ibconexfer --height 3
 impl Runnable for QueryPacketAcknowledgementsCmd {
     fn run(&self) {
+        use crate::conclude::json;
```
Comment (ljoss17) [inline crates/relayer-cli/src/commands/query/packet/acks.rs:68]: Should we move this at the top, with the other `use` calls?

File: crates/relayer-cli/src/commands/query/packet/commitments.rs [hunk 2]
```diff
@@ -67,8 +60,11 @@ impl QueryPacketCommitmentsCmd {
 // cargo run --bin hermes -- query packet commitments --chain ibc-0 --port transfer --channel ibconexfer --height 3
 impl Runnable for QueryPacketCommitmentsCmd {
     fn run(&self) {
+        use crate::conclude::json;
```
Comment (ljoss17) [inline crates/relayer-cli/src/commands/query/packet/commitments.rs:63]: Same here

File: crates/relayer-cli/src/commands/query/packet/pending.rs [hunk 3]
```diff
@@ -96,9 +110,12 @@ impl QueryPendingPacketsCmd {
 
 impl Runnable for QueryPendingPacketsCmd {
     fn run(&self) {
+        use crate::conclude::json;
```
Comment (ljoss17) [inline crates/relayer-cli/src/commands/query/packet/pending.rs:113]: Same here

File: crates/relayer-cli/src/commands/query/packet/pending_acks.rs [hunk 4]
```diff
@@ -76,9 +77,12 @@ impl QueryPendingAcksCmd {
 
 impl Runnable for QueryPendingAcksCmd {
     fn run(&self) {
+        use crate::conclude::json;
```
Comment (ljoss17) [inline crates/relayer-cli/src/commands/query/packet/pending_acks.rs:80]: Same here

File: crates/relayer-cli/src/commands/query/packet/pending_sends.rs [hunk 5]
```diff
@@ -76,9 +77,12 @@ impl QueryPendingSendsCmd {
 
 impl Runnable for QueryPendingSendsCmd {
     fn run(&self) {
+        use crate::conclude::json;
```
Comment (ljoss17) [inline crates/relayer-cli/src/commands/query/packet/pending_sends.rs:80]: Same here

File: crates/relayer-cli/src/commands/query/packet/util.rs [hunk 6]
```diff
@@ -0,0 +1,63 @@
+use core::fmt;
+
+use serde::Serialize;
+
+use ibc_relayer::util::collate::{Collated, CollatedIterExt};
+use ibc_relayer_types::core::ics04_channel::packet::Sequence;
+use ibc_relayer_types::Height;
+
+pub use ibc_relayer::chain::counterparty::PendingPackets;
```
Comment (ljoss17) [inline crates/relayer-cli/src/commands/query/packet/util.rs:9]: I am not sure to understand why the `PendingPackets` is re-exported?

**Review by ljoss17 — COMMENTED** [review]
A few comments/questions regarding imports

File: crates/relayer-cli/src/commands/query/packet/acks.rs (hunk 1 above)
Comment (romac) [reply crates/relayer-cli/src/commands/query/packet/acks.rs:68]: It's only used in this function, so I would keep it here. Local imports for one-time use are fine imho.

File: crates/relayer-cli/src/commands/query/packet/pending.rs (hunk 3 above)
Comment (romac) [reply crates/relayer-cli/src/commands/query/packet/pending.rs:113]: As above

File: crates/relayer-cli/src/commands/query/packet/pending_sends.rs (hunk 5 above)
Comment (romac) [reply crates/relayer-cli/src/commands/query/packet/pending_sends.rs:80]: As above

File: crates/relayer-cli/src/commands/query/packet/pending_acks.rs (hunk 4 above)
Comment (romac) [reply crates/relayer-cli/src/commands/query/packet/pending_acks.rs:80]: As above

File: crates/relayer-cli/src/commands/query/packet/util.rs (hunk 6 above)
Comment (romac) [reply crates/relayer-cli/src/commands/query/packet/util.rs:9]: Just as a convenience, this way one can get both the `PendingPackets` struct and its `Collated` counterpart with one import line.

**Pull request merged**